middleware/allRoutes/handler.go

```diff
@@ -25,29 +23,18 @@ func Handler(routesHandler map[string]http.Handler, zebedeeClient *client.Client
 
 			// Populate context here with language
 			req = dprequest.SetLocaleCode(req)
-
-			if strings.HasSuffix(path, "/latest") {
-				log.Event(req.Context(), "Skipping content specific handling as it's a request to a known URL.",
-					log.INFO, log.Data{"url": req.URL.String()})
-				h.ServeHTTP(w, req)
-				return
-			}
-
-			// No point calling zebedee for these paths so skip middleware
-			if ok, err := regexp.MatchString(`^\/(?:datasets|filter|feedback|healthcheck)`, path); ok && err == nil {
-				log.Event(req.Context(), "Skipping content specific handling as not relevant on this path.", log.INFO, log.Data{"url": path})
-				h.ServeHTTP(w, req)
-				return
+			ok, err := regexp.MatchString(`^(?:data\.json)`, path)
```

Comment:
Personally I dislike Regex. They are a powerful tool but not without their drawbacks - they can be computationally expensive (something to bear in mind with lost of requests hitting this code) and there is a chance it's not catching all the cases you think it is. It can be had to test/prove its working 100% the way you think it is.

For this case I don't think you need to use a regex. `req.URL.Path` will return the URI minus any query parameters. If you use [filepath.Base](https://golang.org/pkg/path/filepath/#Base) to get the last element of the path you can do a simple string comparison:

````
filename := filepath.Base(req.URL.Path)
if filename == "data.json" {
    ....
}
````
This feels simpler, safer and easier to understand/test to me. Additionally it removes and error case you have to handle == less code.

Reply:
Yer that is fair, was in the mindset of regex at the time the file was full of it. I should have taken a step back. definitely, good reasons to use `filepath.Base` over the regex checks.